rust/lance-index/src/vector/v3/subindex.rs

```diff
@@ -60,3 +69,71 @@ pub trait IvfSubIndex<T: Num, Store: VectorStore>: Send + Sync + Sized {
         self.to_array()
     }
 }
+
+struct FlatIndex {}
+
+lazy_static::lazy_static! {
+    static ref ANN_SEARCH_SCHEMA: SchemaRef = Arc::new(Schema::new(vec![
+        Field::new(DIST_COL, DataType::Float32, true),
+        ROW_ID_FIELD.clone(),
+    ]));
+}
+
+impl<T: Num, Store: VectorStore> IvfSubIndex<T, Store> for FlatIndex {
+    type QueryParams = ();
+
+    fn index_name(&self) -> &str {
+        "FLAT"
+    }
+
+    fn search(
+        &self,
+        query: &[T],
+        k: usize,
+        _params: Self::QueryParams,
+        storage: &Store,
+        pre_filter_bitmap: Option<RoaringBitmap>,
+    ) -> Result<RecordBatch> {
+        let dist_calc = storage.dist_calculator_from_native(query);
+        let (row_ids, dists): (Vec<u64>, Vec<f32>) = (0..storage.len())
```

Comment:
Is there a place to write down the contract where `pre_filter_bitmap` contains `0..storage.len()` as ids?

Reply:
hmm, this is my bad. should be filtering for `pre_filter_bitmap .contains(storage.row_ids()[id as usize])`